src/rgw/rgw_rados.cc

```diff
@@ -243,10 +245,12 @@ void RGWObjectCtx::set_atomic(rgw_obj& obj) {
   assert (!obj.empty());
   objs_state[obj].is_atomic = true;
 }
-void RGWObjectCtx::set_prefetch_data(const rgw_obj& obj) {
+void RGWObjectCtx::set_prefetch_data(const rgw_obj& obj, off_t start, off_t end) {
   std::unique_lock wl{lock};
   assert (!obj.empty());
   objs_state[obj].prefetch_data = true;
+  objs_state[obj].start = start;
+  objs_state[obj].end = end;
```

Comment:
don't repeat the std::map lookup for each assignment

Reply:
done